src/outbound.rs

```diff
@@ -173,6 +175,10 @@ impl Outbound {
                 // replaces our buffer
                 self.updated.wait(&mut writer);
             }
+
+            drop(writer);
+
+            std::thread::sleep(before.elapsed());
```

Comment:
Should we sleep here?

Reply:
we want to allow writers to have a chance to batch up their writes a bit, which they won't be able to do if we keep taking out the mutex in a loop. but the issue we want to avoid is sleeping longer than necessary, which would add too much latency to the non-loaded system.

this approach aims to maintain roughly 50% lock time dedicated to the flusher, and 50% of the time available for the user to batch up writes. so, it avoids sleeping too long, and auto-tunes to roughly maintain this proportion of writer-related contention.

Reply:
Should we sleep first thing when kicked?

Reply:
this is roughly based on a batching technique that @utaal mentioned to me for auto-tuning batch sizes in extremely high-throughput distributed dataflow processing systems, while still maintaining nice latency for messages while under low-load

Reply:
if we sleep right after being kicked, we double the mutex-related contention because we double-lock on each acquisition